core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/parser/scanner/CPreprocessor.java

```diff
@@ -352,6 +353,10 @@ public CPreprocessor(FileContent fileContent, IScannerInfo info, ParserLanguage
 		fIncludeSearchPath = configureIncludeSearchPath(new File(contextPath).getParentFile(), info);
 		setupMacroDictionary(configuration, info, language);
 
+		if (fMacroDictionary.containsKey(CPP_IMPL_THREE_WAY_COMPARISON)) {
```

Comment:
I don't know if enabling the feature via a feature macro is how it is "supposed to be done", but it seems to work fine.